testsuite/cluster-test/src/tx_emitter.rs

```diff
@@ -890,62 +892,25 @@ async fn retrieve_account_balance(
         .balances)
 }
 
-pub fn sign_with_transaction_builder(
-    builder: TransactionBuilder,
-    sender_account: &mut AccountData,
-) -> anyhow::Result<SignedTransaction> {
-    let raw_txn = builder
-        .sender(sender_account.address)
-        .sequence_number(sender_account.sequence_number)
-        .build();
-    sender_account.key_pair.sign_txn(raw_txn)
-}
-
-pub fn gen_submit_transaction_request(
-    payload: TransactionPayload,
-    sender_account: &mut AccountData,
-    chain_id: ChainId,
-    gas_price: u64,
-) -> SignedTransaction {
-    let transaction = create_user_txn(
-        &sender_account.key_pair,
-        payload,
-        sender_account.address,
-        sender_account.sequence_number,
-        MAX_GAS_AMOUNT,
-        gas_price,
-        GAS_CURRENCY_CODE.to_owned(),
-        TXN_EXPIRATION_SECONDS,
-        chain_id,
-    )
-    .expect("Failed to create signed transaction");
-    sender_account.sequence_number += 1;
-    transaction
-}
-
 fn gen_mint_request(
-    faucet_account: &mut AccountData,
+    faucet_account: &mut LocalAccount,
     num_coins: u64,
     chain_id: ChainId,
 ) -> SignedTransaction {
-    let receiver = faucet_account.address;
-    let transaction = sign_with_transaction_builder(
+    let receiver = faucet_account.address();
+    faucet_account.sign_with_transaction_builder(
         TransactionFactory::new(chain_id).peer_to_peer_with_metadata(
             Currency::XUS,
             receiver,
             num_coins,
             vec![],
             vec![],
         ),
-        faucet_account,
     )
-    .expect("Failed to create signed transaction");
-    faucet_account.sequence_number += 1;
-    transaction
 }
 
 pub fn gen_transfer_txn_request(
```

Comment:
For this function and the `mint_request` function we're probably going to want to change the function signature to take in a `&TransactionFactory` instead of a chainid so we can reuse a factory that's already been constructed with defaults rather than creating one on the fly every time.